File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IPhasorValue.cs
//*******************************************************************************************************
//  IPhasorValue.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using TVA.Units;

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of a phasor value.
    /// </summary>
    public interface IPhasorValue : IChannelValue<IPhasorDefinition>
    {
        /// <summary>
        /// Gets the <see cref="PhasorProtocols.CoordinateFormat"/> of this <see cref="IPhasorValue"/>.
        /// </summary>
        CoordinateFormat CoordinateFormat { get; }

        /// <summary>
        /// Gets the <see cref="PhasorType"/> of this <see cref="IPhasorValue"/>.
        /// </summary>
        PhasorType Type { get; }

        /// <summary>
        /// Gets or sets the <see cref="TVA.Units.Angle"/> value (a.k.a., the argument) of this <see cref="IPhasorValue"/>, in radians.
        /// </summary>
        Angle Angle { get; set; }

        /// <summary>
        /// Gets or sets the magnitude value (a.k.a., the absolute value or modulus) of this <see cref="IPhasorValue"/>.
        /// </summary>
        double Magnitude { get; set; }

        /// <summary>
        /// Gets or sets the real floating point value of this <see cref="IPhasorValue"/>.
        /// </summary>
        double Real { get; set; }

        /// <summary>
        /// Gets or sets the imaginary floating point value of this <see cref="IPhasorValue"/>.
        /// </summary>
        double Imaginary { get; set; }

        /// <summary>
        /// Gets or sets the unscaled integer representation of the real value of this <see cref="IPhasorValue"/>.
        /// </summary>
        int UnscaledReal { get; set; }

        /// <summary>
        /// Gets or sets the unscaled integer representation of the imaginary value of this <see cref="IPhasorValue"/>.
        /// </summary>
        int UnscaledImaginary { get; set; }
    }
}

File: Source/OldCode/Archive/2004 TVA Code Library/Source/EE/Phasor/IPhasorValue.vb
'*******************************************************************************************************
'  IPhasorValue.vb - Phasor value interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2003
'  Primary Developer: James R Carroll, System Analyst [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Namespace EE.Phasor

    ' This class represents the protocol independent interface of a phasor value.
    Public Interface IPhasorValue

        Inherits IChannelValue

        Shadows Property Definition() As IPhasorDefinition

        ReadOnly Property CoordinateFormat() As CoordinateFormat

        ReadOnly Property [Type]() As PhasorType

        Property Angle() As Double

        Property Magnitude() As Double

        Property Real() As Double

        Property Imaginary() As Double

        Property UnscaledReal() As Int16

        Property UnscaledImaginary() As Int16

    End Interface

End Namespace